setup.py

```diff
@@ -1,15 +1,80 @@
 #!/usr/bin/env python
+import os
+import subprocess
+import sys
+from glob import glob
+from setuptools import setup, find_packages, Extension
 
-try:
-    from setuptools import setup
-except ImportError:
-    from distribute_setup import use_setuptools
-    use_setuptools()
-    from setuptools import setup
+
+if os.path.exists('relic'):
+    sys.path.insert(1, 'relic')
+    import relic.release
+else:
+    try:
+        import relic.release
+    except ImportError:
+        try:
+            subprocess.check_call(['git', 'clone',
+                'https://github.com/jhunkeler/relic.git'])
+            sys.path.insert(1, 'relic')
+            import relic.release
+        except subprocess.CalledProcessError as e:
+            print(e)
+            exit(1)
+
+
+version = relic.release.get_info()
+relic.release.write_template(version, 'lib/reftools')
 
 setup(
-    setup_requires=['d2to1>=0.2.9', 'stsci.distutils>=0.3.2'],
-    d2to1=True,
-    use_2to3=False,
-    zip_safe=False
+    name = 'reftools',
+    version = version.pep386,
```

Comment:
Versions are immutable. If you released 2.0 without a feature, then you have to increment version by either minor or patch. It's still 2.0-ish.

2.0.0 - `foo()` and `bar()`
2.0.1 - `foo2()` deprecated `foo()`
[10 commits later]
2.0.2 - `baz()` added as a wrapper for `foo2()` and `bar()` because it's a nice addition
[30 commits later]
2.1.0 - `new_feature_here()`, `here_too(), `here_there()`

All of them are technically derived from 2.0, but are now easier to track because you can account for features appearing/disappearing over time.

If I tell the user, yeah we totally broke functionality of `bar()` in 2.0.3, (so I look at the history and see the last unchanged `bar()` is derived from tag `2.0.2` plus 4 commits).

**Two options:**

1. Tag the pre-bugged code
`git tag -a 2.0.2a -m 'Before bar() bug' [hash of 2.0.2+4]`
`git push --tags`

The user then builds from tag 2.0.2a as a dirty hotfix.

2. Fix `bar()` upstream and release 2.0.4 (probably should do that anyway)